blockdag/reachability.go

```diff
@@ -382,6 +617,23 @@ func (rtn *reachabilityTreeNode) isAncestorOf(other *reachabilityTreeNode) bool
 	return rtn.interval.isAncestorOf(other.interval)
 }
 
+// findCommonAncestor finds the most recent reachability tree ancestor
+// common to both rtn and other.
+func (rtn *reachabilityTreeNode) findCommonAncestor(other *reachabilityTreeNode) *reachabilityTreeNode {
+	currentThis := rtn
+	currentOther := other
+	for {
+		if currentThis.parent == nil || currentThis.isAncestorOf(other) {
```

Comment:
the test for nil is redundant because genesis will always return true for `.isAncestorOf(other)`
(same goes for 3 lines below)